src/pages/SignIn.js

```diff
@@ -1,47 +1,81 @@
 import {useContext, useState} from 'react';
 import {useNavigate} from 'react-router-dom';
 import UserCtx from '../context/user-context';
+import EnhancedInput from '../Helpers/EnhancedInput';
+import validations from '../Helpers/Validations';
 
 const SignIn = props => {
 
     const userCtx = useContext(UserCtx);
     const navigate = useNavigate();
     const [userData, setUserData] = useState({
-        userName: '',
+        username: '',
         password: ''
     });
-    
+
+    const [fieldHasError, setFieldHasError] = useState({
+        username: false,
+        password: false
+    });
+
+    const emailValidations = [validations.required, validations.emailPattern];
+    const passwordValidations = [validations.required, validations.minLength,validations.passwordPattern];
+
+    const inputsData = [
+        {
+            type: "text", 
+            id: "username", 
+            label: "User name",
+            rules: emailValidations,
+            autocomplete: "off"
+        },
+        {
+            type: "password", 
+            id: "password", 
+            label: "Password",
+            rules: passwordValidations,
+            autocomplete: "on"
+        },
+    ]
```

Comment:
You have 3 states/configs objects for the same fields. Why so? You can combine them to the single state.